pallets/payment/src/lib.rs

```diff
@@ -91,15 +96,38 @@ pub mod pallet {
 		/// The only action is to store the details of this payment in storage and reserve
 		/// the specified amount.
 		#[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
-		pub fn create(
+		pub fn pay(
+			origin: OriginFor<T>,
+			recipient: T::AccountId,
+			asset: AssetIdOf<T>,
+			amount: BalanceOf<T>,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+			<Self as PaymentHandler<T::AccountId, AssetIdOf<T>, BalanceOf<T>>>::create_payment(
+				who, recipient, asset, amount, None,
+			)?;
+			Ok(().into())
+		}
+
+		/// This allows any user to create a new payment with the option to add a remark, this remark
+		/// can then be used to run custom logic and trigger alternate payment flows.
+		/// the specified amount.
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+		pub fn pay_with_remark(
 			origin: OriginFor<T>,
 			recipient: T::AccountId,
 			asset: AssetIdOf<T>,
 			amount: BalanceOf<T>,
+			remark: Vec<u8>,
 		) -> DispatchResultWithPostInfo {
 			let who = ensure_signed(origin)?;
+			// TODO : apply custom logic based on remark
```

Comment:
Should we have some custom logic to implement here right now or we do that in future? 🤔

Reply:
We already have some form of custom user-provided logic with the `FeeHandler`, so one thing we can do is to allow the `apply_fees` method to receive a reference to the `PaymentDetail` that contains the remark